### DISCUSSION TITLE
To sychronize user list from IAM?

### DISCUSSION THREAD

**Discussion by zhangxt-ihep** [DIRACGrid/DIRAC#7558] [Q&A] [answered]
Hello,

  It seems that VOMSAA in IAM doesn't have the same VOMS API where VOMS2CS can use to import user list. Is their any DIRAC agent that can synchronize user list from IAM working like VOMS2CSAgent?

Xuantong

**Comment by chaen** [comment]
Hi.
That's a (bad) surprise. I did not try yet. Does it have a different API or is it just inaccessible ?

**Comment by chaen** [comment]
It appears you are right ! 
Would you be willing to implement the necessary changes in the VOMS2CS agent, or probably a IAM2CS agent ?
Here are useful resources:
https://twiki.cern.ch/twiki/bin/view/LCG/TokenSupportDevTips#How_to_get_user_identities_from
https://indigo-iam.github.io/v/v1.8.3/docs/reference/api/scim-api/

**Comment by zhangxt-ihep** [comment]
Hi @chaen ,

I think I could try to add a new IAM2CS agent to sychronize user list, but I am busy at other work, so it could take some time to finish it.

Xuantong

**Reply by chaen** [reply, marked as answer]
Thanks ! FYI, there is already an issue for that with most of the code you'll need

https://github.com/DIRACGrid/DIRAC/issues/7416

**Reply by zhangxt-ihep** [reply]
I found you have already finished the development and committed. Thank you. When could it be released?

**Reply by chaen** [reply]
There are still teething issues with VOMS-IAM synchronizer scripts ran by the IAM team, so it will be released once these are fixed

**Reply by marianne013** [reply]
@chaen Which issue are you referring to ?

**Reply by chaen** [reply]
That one which was merged https://github.com/indigo-iam/voms-importer/pull/22
and there are still weird cases that Berk is investigating
